go/vt/vttablet/tabletserver/planbuilder/plan.go

```diff
@@ -324,31 +308,20 @@ func BuildMessageStreaming(name string, tables map[string]*schema.Table) (*Plan,
 	return plan, nil
 }
 
-// checkForPoolingUnsafeConstructs returns an error if the SQL expression contains
-// a call to GET_LOCK(), which is unsafe with server-side connection pooling.
-// For more background, see https://github.com/vitessio/vitess/issues/3631.
-func checkForPoolingUnsafeConstructs(expr sqlparser.SQLNode) error {
-
-	genError := func(node sqlparser.SQLNode) error {
-		return vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "%s not allowed without a reserved connections", sqlparser.String(node))
-	}
-
-	if _, isSetStmt := expr.(*sqlparser.Set); isSetStmt {
-		return genError(expr)
-	}
-
-	sel, isSel := expr.(*sqlparser.Select)
-	if !isSel {
-		return nil
-	}
-	return sqlparser.Walk(func(in sqlparser.SQLNode) (kontinue bool, err error) {
+// hasLockFunc looks for get_lock function in the select query.
+// If it is present then it returns true otherwise false
+func hasLockFunc(sel *sqlparser.Select) bool {
+	var found bool
+	_ = sqlparser.Walk(func(in sqlparser.SQLNode) (bool, error) {
 		lFunc, isLFunc := in.(*sqlparser.LockingFunc)
 		if !isLFunc {
 			return true, nil
 		}
 		if lFunc.Type == sqlparser.GetLock {
-			return false, genError(lFunc)
+			found = true
+			return false, nil
 		}
```

Comment:
Shouldn't we also include `ReleaseAllLocks`?

Reply:
only get_lock should happen with reserved conn, otherwise it does not matter.

Reply:
got it